src/main/java/fi/vm/yti/terminology/api/frontend/FrontendController.java

```diff
@@ -164,22 +166,28 @@ JsonNode getVocabularyList(@RequestParam(required = false, defaultValue = "true"
     @io.swagger.v3.oas.annotations.parameters.RequestBody(description = "The JSON data for the new terminology node")
     @ApiResponse(responseCode = "200", description = "The ID for the newly created terminology")
     @PostMapping(path = "/vocabulary", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
-    UUID createVocabulary(@Parameter(description = "The meta model graph for the new terminology") @RequestParam UUID templateGraphId,
+    String createVocabulary(@Parameter(description = "The meta model graph for the new terminology") @RequestParam UUID templateGraphId,
                           @Parameter(description = "The prefix, i.e., freely selectable part of terminology namespace") @RequestParam String prefix,
                           @Parameter(description = "If given, tries to use the ID for the terminology") @RequestParam(required = false) @Nullable UUID graphId,
                           @Parameter(description = "Whether to do synchronous creation, i.e., wait for the result. This is recommended.") @RequestParam(required = false, defaultValue = "true") boolean sync,
-                          @RequestBody GenericNode vocabularyNode) {
+                          @Parameter(description = "Do not actually create vocabulary, only validate the input.") @RequestParam(required = false, defaultValue = "false") boolean validateOnly,
+                          @ValidVocabularyNode @RequestBody GenericNode vocabularyNode) {
 
         try {
             logger.info("POST /vocabulary requested with params: templateGraphId: " +
                 templateGraphId.toString() + ", prefix: " + prefix + ", vocabularyNode.id: " + vocabularyNode.getId().toString());
 
             UUID predefinedOrGeneratedGraphId = graphId != null ? graphId : UUID.randomUUID();
-            termedService.createVocabulary(templateGraphId, prefix, vocabularyNode, predefinedOrGeneratedGraphId, sync);
-            logger.debug("Vocabulary with prefix \"" + prefix + "\" created");
-            return predefinedOrGeneratedGraphId;
+
+            if (!validateOnly) {
+                termedService.createVocabulary(templateGraphId, prefix, vocabularyNode, predefinedOrGeneratedGraphId, sync);
+                logger.debug("Vocabulary with prefix \"" + prefix + "\" created");
+                return predefinedOrGeneratedGraphId.toString();
+            }
+
+            return "OK";
```

Comment:
Is it clearer to create separate endpoint for validation, e.g. POST /vocabulary/validate?

Reply:
yes, done